packages/amplify-category-function/src/provider-utils/awscloudformation/service-walkthroughs/execPermissionsWalkthrough.ts

```diff
@@ -155,18 +155,15 @@ export const askExecRolePermissionsQuestions = async (
         }
       }
     } catch (e) {
-      if (e.name === 'MethodNotFound') {
+      if (e.name === 'PluginMethodNotFoundError') {
         context.print.warning(`${selectedCategory} category does not support resource policies yet.`);
+        context.usageData.emitError(e);
```

Comment:
I don't think this should emit an error here because we're not done executing.

```suggestion
```

Reply:
I am open to remove it, kept it since it was previously there for both branches